weave.lisp

```diff
@@ -197,71 +227,110 @@
-                    (setf (weaver-title weaver)
-                          (second expr)))
-                  (incf (weaver-chapter-counter weaver))
-                  (setf (weaver-section-counter weaver) -1)
-                  (format t "<h1>~a<a id=\"~a\"></a></h1>~%" 
-                          (second expr)
-                          (chapter-id (weaver-chapter-counter weaver))))
-                 (:S
-                  (incf (weaver-section-counter weaver))
-                  (format t "<h2>~a. ~a<a id=\"~a\"></a></h2>~%" 
-                          (+ (weaver-section-counter weaver) 1)
-                          (second expr)
-                          (section-id 
-                            (weaver-section-counter weaver)
-                            (weaver-chapter-counter weaver))))
-                 (:CODE_TYPE
-                  (let* ((args (split-whitespace (second expr)))
-                         (language (first args))
-                         (extension (subseq (second args) 1)))
-                    (setf (gethash extension (weaver-code-type-table weaver)) language)
-                    (push extension (weaver-used-extensions weaver))))
-                 (:COMMENT_TYPE nil)
-                 (:ADD_CSS nil)
-                 (:OVERWRITE_CSS nil)
-                 (:COLORSCHEME nil)
-                 (:ERROR_FORMAT nil)
-                 (:MATHBLOCK
-                  ; Put <code> tags in block so it displays tex nicely without JS.
-                  (setf (weaver-used-math weaver) t)
-                  (write-string "<div class=\"math-block\"><code>")
-                  (when (not (equal (second expr) "displaymath"))
-                      (format t "\\begin{~a}" (second expr)))
-                  (write-separated-list (third expr) #\newline *standard-output*)
-                  (when (not (equal (second expr) "displaymath"))
-                      (format t "\\end{~a}" (second expr)))
-                  (write-string "</code></div>"))
-                 (:MATH
-                  ; Use backticks to prevent markdown from formatting tex,
-                  ; for example treating _ as emphasis. 
-                  (setf (weaver-used-math weaver) t)
-                  (format t "<span class=\"math\">`~a`</span>"
-                          (second expr)))
-                 (:TOC (weave-toc
-                         (weaver-toc weaver)
-                         (textblockdef-file def)))
-                 (otherwise (error 'user-error
-                                   :format-control "unknown prose command ~S"
-                                   :format-arguments (first expr)))))
-              (t (error "unknown structure ~s" expr)))))
+    (cond ((stringp expr)
+           (if (ppcre:scan *anchor-pattern* expr)
+               (write-string
+                (replace-sections-with-id-anchors
+                 (weaver-section-map weaver)
+                 expr))
+               (write-string expr)))
```

Comment:
^ This is the new bit that is obscured by all of the blank space changes.

Reply:
Any regex scanning like this should happen in the parser. By the time it gets here it should be a structured command.
For example, all the `here is a line with a @{reference}` becomes `"here is a line with a " (:INCLUDE "reference")`. So similarly, this should be a command type like `(:SECTION-REFERENCE "name")`.